confmap/provider.go

```diff
@@ -112,21 +112,29 @@ type retrievedSettings struct {
 }
 
 // RetrievedOption options to customize Retrieved values.
-type RetrievedOption func(*retrievedSettings)
+type RetrievedOption interface {
```

Comment:
Not a breaking change because users could not create `func(*retrievedSettings)` types anyway because `retrievedSettings` was private.